File: src/main/java/com/nndmove/app/service/dto/PlaylistDTO.java
package com.nndmove.app.service.dto;

import java.io.Serializable;
import java.util.Objects;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * A DTO for the {@link com.nndmove.app.domain.Playlist} entity.
 */
@SuppressWarnings("common-java:DuplicatedBlocks")
@Getter
@Setter
@NoArgsConstructor
public class PlaylistDTO implements Serializable {

    private Long id;

    private UserDTO user;

    private MovieDTO movie;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof PlaylistDTO)) {
            return false;
        }

        PlaylistDTO playlistDTO = (PlaylistDTO) o;
        if (this.id == null) {
            return false;
        }
        return Objects.equals(this.id, playlistDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "PlaylistDTO{" +
            "id=" + getId() +
            ", user=" + getUser() +
            ", movie=" + getMovie() +
            "}";
    }
}

File: src/main/java/com/nndmove/app/service/dto/PlaylistDTO.kt
package com.nndmove.app.service.dto

import java.io.Serializable
import java.util.*

/**
 * A DTO for the [com.nndmove.app.domain.Playlist] entity.
 */
class PlaylistDTO : Serializable {
    @JvmField
    var id: Long? = null

    @JvmField
    var user: UserDTO? = null

    @JvmField
    var movie: MovieDTO? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is PlaylistDTO) {
            return false
        }

        if (this.id == null) {
            return false
        }
        return this.id == o.id
    }

    override fun hashCode(): Int {
        return Objects.hash(this.id)
    }

    // prettier-ignore
    override fun toString(): String {
        return "PlaylistDTO{" +
                "id=" + id +
                ", user=" + user +
                ", movie=" + movie +
                "}"
    }
}
